.github/workflows/test.yml

```diff
@@ -0,0 +1,183 @@
+  # Prevent the build from running when there are only irrelevant changes.
+  push:
+    paths-ignore:
+      - '**.md'
+      - '**.txt'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+      - '**.txt'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Notes regarding supported versions in WP:
+        # PHP 5.3-5.5 are supported since WP 3.7 until WP 5.1 (including).
+        # PHP 5.6 is supported since WP 4.1.
+        # PHP 7.0 is supported since WP 4.4.
+        # PHP 7.1 is supported since WP 4.7.
+        # PHP 7.2 is supported since WP 4.9.
+        # PHP 7.3 is supported since WP 5.0.
+        # PHP 7.4 is supported since WP 5.3.
+        # PHP 8.0 is sort of supported since WP 5.6.
+        # This doesn't mean that the plugin doesn't work on earlier version as it may not hit incompatible
+        # parts of WP, but just something to be aware of in case of failures.
+        php: ['5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0']
+        wp: ['latest', '3.7']
+        experimental: [false]
+
+        include:
+          # As support for PHP < 5.6 was dropped in WP 5.2, PHP 5.3 - 5.5 can only be tested against WP < 5.2.
+          # Complement the builds run via the matrix with high/low WP builds for PHP 5.3 - 5.5.
+          - php: '5.5'
+            wp: '5.1'
+            experimental: false
+          - php: '5.5'
+            wp: '3.7'
+            experimental: false
+          - php: '5.4'
+            wp: '5.1'
+            experimental: false
+          - php: '5.4'
+            wp: '3.7'
+            experimental: false
+          - php: '5.3'
+            wp: '5.1'
+            experimental: false
+          - php: '5.3'
+            wp: '3.7'
+            experimental: false
+
+          # Complement the builds run via the matrix with some additional builds against specific WP versions.
+          - php: '7.4'
+            wp: '5.4'
+            experimental: false
+          - php: '7.3'
+            wp: '5.3'
+            experimental: false
+          - php: '7.2'
+            wp: '5.2'
+            experimental: false
+          - php: '7.1'
+            wp: '4.6'
+            experimental: false
+          - php: '7.0'
+            wp: '4.3'
+            experimental: false
+          - php: '5.6'
+            wp: '4.0'
+            experimental: false
+          - php: '5.6'
+            wp: '5.5'
+            experimental: false
+          - php: '5.5'
+            wp: '4.5'
+            experimental: false
+          - php: '5.4'
+            wp: '5.0'
+            experimental: false
+          - php: '5.3'
+            wp: '4.9'
+            experimental: false
+
+          # Experimental builds. These are allowed to fail.
+          - php: '7.4'
+            wp: 'trunk'
+            experimental: true
+
+          - php: '8.1'
+            wp: 'trunk'
+            experimental: true
+
+    name: "Test: PHP ${{ matrix.php }} - WP ${{ matrix.wp }}"
```

Comment:
Don't feel strongly about this, but it always feels repetitive when naming jobs to include the name of the workflow in the job name. Test is implied here from the workflow name.

Reply:
Fair enough. As this plugin doesn't have separate "quick test" or code coverage runs which could be using a similar name pattern, having the added `Test` is unnecessary.

I would note though that as soon as either of these are added, a keyword to distinguish between the various test run times would be needed, but that's for later (if at all).